test-harness/src/test/groovy/com/netflix/conductor/test/integration/JsonJQTransformSpec.groovy

```diff
@@ -84,4 +85,66 @@ class JsonJQTransformSpec extends AbstractSpecification {
             tasks[0].reasonForIncompletion as String == "Cannot index string with string \"array\""
         }
     }
+
+    /**
+     * Given the following invalid input JSON
+     *{*   "in1": "a",
+     *   "in2": "b"
+     *}* using the same query from the success test, jq will try to get in1['array']
+     * and fail since 'in1' is a string.
+     *
+     * Re-run failed system task with the following valid input JSON will fix the workflow
+     *{*   "in1": {*     "array": [ "a", "b" ]
+     *},
+     *   "in2": {*     "array": [ "c", "d" ]
+     *}*}* expect the workflow task to transform to following result:
+     *{*     out: [ "a", "b", "c", "d" ]
+     *}
+     */
+    def "Test rerun workflow with failed json jq transform task"() {
+        given: "workflow input"
+        def invalidInput = new HashMap()
+        invalidInput['in1'] = "a"
+        invalidInput['in2'] = "b"
+
+        def validInput = new HashMap()
+        def input = new HashMap()
+        input['in1'] = new HashMap()
+        input['in1']['array'] = ["a", "b"]
+        input['in2'] = new HashMap()
+        input['in2']['array'] = ["c", "d"]
+        validInput['input'] = input
+        validInput['queryExpression'] = '.input as $_ | { out: ($_.in1.array + $_.in2.array) }'
+
+        when: "workflow which has the json jq transform task started"
+        def workflowInstanceId = workflowExecutor.startWorkflow(JSON_JQ_TRANSFORM_WF, 1,
+                '', invalidInput, null, null, null)
+
+        then: "verify that the workflow and task failed with expected error"
+        with(workflowExecutionService.getExecutionStatus(workflowInstanceId, true)) {
+            status == Workflow.WorkflowStatus.FAILED
+            tasks.size() == 1
+            tasks[0].status == Task.Status.FAILED
+            tasks[0].taskType == 'JSON_JQ_TRANSFORM'
+            tasks[0].reasonForIncompletion as String == "Cannot index string with string \"array\""
```

Comment:
The `as String` may not be necessary. Also, using single quotes `'` ensures that Groovy uses a `String` and not a `GString`.

Reply:
Thanks, updated!